### ISSUE TITLE
Windows tool_integration_tests_3_6 flakes on `expression_evaluation_test`

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#103215]
(part of https://github.com/flutter/flutter/issues/97036)

Hit one flake today:

 [expression_evaluation_test.dart](https://ci.chromium.org/ui/p/flutter/builders/prod/Windows%20tool_integration_tests_3_6/1293/overview)
```
03:23 +9 -1: test\integration.shard\expression_evaluation_test.dart: flutter test expression evaluation - can evaluate expressions returning complex objects in a test [E]                             
  Expected: <Instance of 'InstanceRef'> with `valueAsString`: '2022-5-5'
    Actual: InstanceRef:<[InstanceRef id: objects/5, kind: String, identityHashCode: 0, classRef: [ClassRef id: classes/91, name: _OneByteString, library: [LibraryRef id: libraries/@0150898, name: dart.core, uri: dart:core]]]>
     Which: has `valueAsString` with value '2022-5-6' which is different.
            Expected: 2022-5-5
              Actual: 2022-5-6
                             ^
             Differ at offset 7
  
  package:test_api                                              expect
  test\integration.shard\expression_evaluation_test.dart 189:3  expectValue
  test\integration.shard\expression_evaluation_test.dart 172:3  evaluateComplexReturningExpressions
  ===== asynchronous gap ===========================
  test\integration.shard\expression_evaluation_test.dart 141:5  batch2.<fn>
```

**Comment by zanderso** [comment]
@DanTup The test run straddled midnight.

**Comment by zanderso** [comment]
Marking P4 since this will certainly flake way below the threshold, but should be cleaned up.

**Comment by DanTup** [comment]
> @DanTup The test run straddled midnight.

🙈

Fix at https://github.com/flutter/flutter/pull/103269 - thanks!

**Comment by DanTup** [comment]
Fixed by 4ab68bd04dcc9ecfb74243e8fc324e45fa904774.
